### ISSUE TITLE
Internal error while parsing EventSub Prediction event 

### ISSUE THREAD

**Issue by VGorku** [Teekeks/pyTwitchAPI#312]
Subscriptions and two simple callbacks:

```
async def test_event(data: ChannelPredictionEvent):
    print('EVENT', data.event.broadcaster_user_id)

async def test_end_event(data: ChannelPredictionEndEvent):
    print('EVENT', data.event.broadcaster_user_id)

subscription_ids['begin'] = await eventsub.listen_channel_prediction_begin(subscription_row['twitch_id'], test_event)
subscription_ids['end'] = await eventsub.listen_channel_prediction_end(subscription_row['twitch_id'], test_end_event)
subscription_ids['lock'] = await eventsub.listen_channel_prediction_lock(subscription_row['twitch_id'], test_event)
```

What I did is that simply created Prediction on Twitch (everything was fine) and closed submissions right after that. Seems like it's failed upon parsing the Lock event https://dev.twitch.tv/docs/eventsub/eventsub-subscription-types/#channelpredictionlock

Sorry for the screenshot, that's all I got for now
![image](https://github.com/Teekeks/pyTwitchAPI/assets/134862354/334a2aa1-caf3-4330-af43-4312b865fc64)

Any suggestions on how to fix that? Thanks <3

**Comment by VGorku** [comment]
I think I got it.

I was testing off-stream, basically starting Predictions and ending them without any vote. It seems that current implementation does not expect data['event']['outcomes']['top_predictors'] to be empty list:
![image](https://github.com/Teekeks/pyTwitchAPI/assets/134862354/66cb9eba-25d4-4e26-bf12-dbb8762ebeea)

Here is my disgusting workaround in twitchAPI/eventsub/webhook.py if someone interested (I dont need top predictors for my project)
![image](https://github.com/Teekeks/pyTwitchAPI/assets/134862354/9fa1f930-b671-427e-b11e-81df62f19eae)

Hope this will be fixed 🤞

**Comment by Teekeks** [comment]
I cant reproduce this by starting a prediction & then either letting it run out or closing it manually without anyone voting.

Could you please provide the following for further investigation:
- Your used Python Version
- The Version of the Library you are using
- An exact step by step guide of how to reproduce this
- A code example that will reproduce this

**Comment by Gregorein** [comment]
@Teekeks, I think this may be related to my issue, as I'm also having problems with `TopPredictors` being handled incorrectly in a `prediction.end` event

 https://github.com/Teekeks/pyTwitchAPI/issues/315

**Comment by Teekeks** [comment]
Fix for this is now available on pypi in v4.2.1. Was not able to reproduce because I already fixed that locally...
